src/Discord.Net/Entities/Webhooks/IWebhook.cs

```diff
@@ -0,0 +1,25 @@
+// todo: docs
+using System.Threading.Tasks;
+
+namespace Discord
+{
+    public interface IWebhook
+    {
+        ValueTask<IUser> GetCreatorAsync(StateBehavior stateBehavior = default, RequestOptions? options = default);
+        ValueTask<ITextChannel> GetChannelAsync(StateBehavior stateBehavior = default, RequestOptions? options = default);
+        ValueTask<IGuild> GetGuildAsync(StateBehavior stateBehavior = default, RequestOptions? options = default);
+        // review: does this need to be an async retrieval? the only path to get an IWebhook should be via an ITextChannel
+
+        string Token { get; }
+
+        // note: yeah, the full creator is included in the payload, but we don't want entities to hold direct references to other entities
+        ulong CreatorId { get; } 
+        ulong ChannelId { get; }
+        ulong GuildId { get; } // review: this is impl'd as a ulong? in d.net legacy, todo-ensure there is no case we wouldn't have the guild ID
```

Comment:
This sounds eerily familiar.

After looking back through our issue tracker...

We had a specific issue with an NRE coming out of webhook users, when trying to access `.GuildId`. Tracked it down to the fact that both `PRESENCE_UPDATE` and `GUILD_MEMBER_UPDATE` were being presented through `DiscordSocketClient` as `GuildMemberUpdated`. The problematic part was that `PRESENCE_UPDATE` by definition, does not include the full details of the user, just its ID, so the old client attempts to grab the full user from cache to present through `GuildMemberUpdated`. For webhook users, however, there would never be any cache entry, and the client would just build up an `IWebhookUser` with the `Guild` left as `null`.

I would definitely support making this not-nullable, if possible.